cmd/influxd/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,69 @@
+package upgrade
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/influxdata/influxdb/v2/bolt"
+)
+
+func TestPathValidations(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	v1Dir := filepath.Join(tmpdir, "v1db")
+	v2Dir := filepath.Join(tmpdir, "v2db")
+
+	boltPath := filepath.Join(v2Dir, bolt.DefaultFilename)
+	enginePath := filepath.Join(v2Dir, "engine")
+
+	err = os.MkdirAll(filepath.Join(enginePath, "db"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	largs := make([]string, 0, 9)
+	largs = append(largs, "--username", "my-user")
+	largs = append(largs, "--password", "my-password")
+	largs = append(largs, "--org", "my-org")
+	largs = append(largs, "--bucket", "my-bucket")
+	largs = append(largs, "--retention", "7d")
+	largs = append(largs, "--token", "my-token")
+	largs = append(largs, "--v1-dir", v1Dir)
+	largs = append(largs, "--bolt-path", boltPath)
+	largs = append(largs, "--engine-path", enginePath)
+	largs = append(largs, "--config-file", "")
+
+	cmd := NewCommand()
+	cmd.SetArgs(largs)
+
+	if err := cmd.Execute(); err == nil {
+		t.Fatal("Must fail")
+	} else {
+		assert.Contains(t, err.Error(), "1.x metadb error")
+	}
+
+	err = os.MkdirAll(filepath.Join(v1Dir, "meta"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(v1Dir, "meta", "meta.db"), []byte{1}, 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cmd = NewCommand()
+	cmd.SetArgs(largs)
+
+	if err := cmd.Execute(); err == nil {
+		t.Fatal("Must fail")
+	} else {
+		assert.Contains(t, err.Error(), "target engine path")
+	}
```

Comment:
You can additionally import "github.com/stretchr/testify/require" and then use:

```suggestion
	err = cmd.Execute()
	require.Error(t, err)
	assert.Contains(t, err.Error(), "target engine path")
```